packaging/amazon-linux-ami-integrated/ecs.service

```diff
@@ -25,6 +25,7 @@ Restart=on-failure
 RestartPreventExitStatus=5
 RestartSec=10s
 EnvironmentFile=-/etc/ecs/ecs.config
+ExecStartPre=/usr/bin/systemctl -q is-active docker
```

Comment:
Can this be verified that this will not be 5 for inactive? Because of `RestartPreventExitStatus=5` policy

Reply:
On the following [page](https://www.freedesktop.org/software/systemd/man/latest/systemd.exec.html#Process%20Exit%20Codes), it looks like it could IF "The program is not installed."? However, this should have been hit in the case where I tried with a vanilla AL AMI where docker isn't installed. For reference, this is the specific exit code:

```
Jul 03 16:23:25 ip-172-31-46-143.us-west-2.compute.internal systemd[1]: ecs.service: control process exited, code=exited status=3
```

Reply:
I guess, we can just always exit with 1 (or just any other number other than 5)

Reply:
Will do something along the following: `/bin/bash -c 'if [ $(/usr/bin/systemctl is-active docker) != "active" ]; then exit 1; fi'`